src/fromager/resolver.py

```diff
@@ -37,59 +34,6 @@
 SUPPORTED_TAGS = set(sys_tags())
 
 
-class Candidate:
-    def __init__(
-        self,
-        name: str,
-        version: Version,
-        url: str | None = None,
-        extras: dict | None = None,
-        is_sdist: bool | None = None,
-    ):
-        self.name = canonicalize_name(name)
-        self.version = version
-        self.url = url
-        self.extras = extras
-        self.is_sdist = is_sdist
-
-        self._metadata = None
-        self._dependencies = None
-
-    def __repr__(self) -> str:
-        if not self.extras:
-            return f"<{self.name}=={self.version}>"
-        return f"<{self.name}[{','.join(self.extras)}]=={self.version}>"
-
-    @property
-    def metadata(self):
-        if self._metadata is None:
-            self._metadata = get_metadata_for_wheel(self.url)
-        return self._metadata
-
-    @property
-    def requires_python(self) -> bool:
-        return self.metadata.get("Requires-Python")
```

Comment:
Removed this function because IMO it is not being used anywhere even within the resolvelib:
fromager: https://github.com/search?q=repo%3Apython-wheel-build%2Ffromager%20requires_python&type=code
resolvelib: https://github.com/search?q=repo%3Asarugaku%2Fresolvelib%20requires_python&type=code

Reply:
Let's keep it in.

A resolver should take Requires-Python into account and fall back to an older version of a package.